lib/grandpa/message_tracker.go

```diff
@@ -122,3 +129,32 @@ func (t *tracker) handleBlock(b *types.Block) {
 		delete(t.commitMessages, h)
 	}
 }
+
+func (t *tracker) handleTick() {
+	t.mapLock.Lock()
+	defer t.mapLock.Unlock()
+
+	for _, vms := range t.voteMessages {
+		for _, v := range vms {
+			// handleMessage would never error for vote message
+			_, err := t.handler.handleMessage(v.from, v.msg)
+			if err != nil {
+				logger.Debugf("failed to handle vote message %v: %s", v, err)
+			}
+
+			if v.msg.Round < t.handler.grandpa.state.round && v.msg.SetID == t.handler.grandpa.state.setID {
+				delete(t.voteMessages, v.msg.Message.Hash)
+			}
+		}
+	}
+
+	for _, cm := range t.commitMessages {
```

Comment:
could we split these loops to run in parallel?

Reply:
Why? Does each iteration take a significant amount of time? :thinking: